pkg/api/api.go

```diff
@@ -515,7 +519,22 @@ func (hs *HTTPServer) registerRoutes() {
 
 		// metrics
 		// DataSource w/ expressions
-		apiRoute.Post("/ds/query", requestmeta.SetSLOGroup(requestmeta.SLOGroupHighSlow), authorize(ac.EvalPermission(datasources.ActionQuery)), routing.Wrap(hs.QueryMetricsV2))
+		dsQueryHandler := routing.Wrap(hs.QueryMetricsV2)
+		if hs.Features.IsEnabledGlobally(featuremgmt.FlagKubernetesQueryServiceRewrite) {
+			// DEV ONLY FEATURE FLAG!
+			// rewrite requests from /ds/query to the new query service
+			namespaceMapper := request.GetNamespaceMapper(hs.Cfg)
+			dsQueryHandler = func(w http.ResponseWriter, r *http.Request) {
+				user, err := appcontext.User(r.Context())
+				if err != nil || user == nil {
+					w.Write([]byte("no user... ???"))
+					return
+				}
+				r.URL.Path = "/apis/query.grafana.app/v0alpha1/namespaces/" + namespaceMapper(user.OrgID) + "/query"
+				hs.clientConfigProvider.DirectlyServeHTTP(w, r)
+			}
+		}
+		apiRoute.Post("/ds/query", requestmeta.SetSLOGroup(requestmeta.SLOGroupHighSlow), authorize(ac.EvalPermission(datasources.ActionQuery)), dsQueryHandler)
```

Comment:
Nit. Rather than implementing this here in api.go might want to do it in metrics.go instead?

Reply:
NOTE: this is likely the last thing we should do -- but is also the easiest way to check if thigns work :)

Long term metrics.go should disappear, and everything it does gets replaced by a query service.  This flag jumps the gun since obviously LOTS missing from the query service still